File: backend/models/Speaker.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db.config');

const Speaker = sequelize.define('Speaker', {
  name: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  bio: {
    type: DataTypes.TEXT,
    allowNull: false,
  },
  profilePicture: {
    type: DataTypes.STRING,
    allowNull: true,
  },
}, {
  timestamps: true,
});

module.exports = Speaker;

File: frontend/lib/models/speaker.dart
class Speaker {
  final int id;
  final String name;
  final String bio;
  final String profilePicture;

  Speaker({
    required this.id,
    required this.name,
    required this.bio,
    required this.profilePicture,
  });

  factory Speaker.fromJson(Map<String, dynamic> json) {
    return Speaker(
      id: json['id'] ?? 0,
      name: json['name'],
      profilePicture: json['profilePicture'],
      bio: json['bio'],
    );
  }
}
